### DISCUSSION TITLE
Licensed add-on plans?

### DISCUSSION THREAD

**Discussion by yohplala** [nautechsystems/nautilus_trader#316] [General]
Hi @cjdsellers and Nautilus trader team,
Thanks for sharing your work on this algo trading platform.
I could read regarding the backtesting result visualization that this would be part of a licensed add-on.

Please, can you share more generally what are your plans as to what will be provided under paid license and what will remain open source?
For instance,
- between CCXT (open source) and CCXT Pro (paid license), CCXT Pro adds websocket connection.
- jesse-trade, another algo trading platform, the platform is open source as well, and the home-made websocket connection to exchanges for carrying out transactions is provided against paid license (so for pure backtest activities, jesse-trade can be freely used, but for live trading, user has to pay a license)

Please, can you share what are your plans for Nautilus trader?
Thanks in advance for sharing your thoughts about this.
Have a good day,
Bests,

**Comment by cjdsellers** [comment]
Thanks for your question!

What will be included in a paid-for add-on will likely include front end type components for visualization and control, as well as monitoring, additional database technologies and other tooling which a more serious trading operation may require. This is however yet to be completely decided.

What is important to note however is that everything which is currently open-source will stay open-source forever.

I'll release some more detailed information on the above in due course.
